FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c

```diff
@@ -2182,7 +2182,8 @@ aid though to optimise the calculations. */
 xUnion32 xSum2, xSum, xTerm;
 xUnionPtr xSource;
 xUnionPtr xLastSource;
-uint32_t ulAlignBits, ulCarry = 0UL;
+intptr_t uxAlignBits;
```

Comment:
I noticed that you have used `intptr_t` here. While it is generally okay to do that, I would suggest exploring `BaseType_t`. Is `BaseType_t` defined for your platform (I know that it is there for all 16 and 32 bit platforms)?

Reply:
Hi @AniruddhaKanhere 

Thanks for the feedback.

I have BaseType_t as an int8_t on my platform but pointers are 16bits.

I also have TickType_t which can be uint16_t or uint32_t depending on configUSE_16_BIT_TICKS but it didn't seem appropriate to use that for a pointer. It'll work, but a 32bit TickType_t will still generate a compiler warning.

What do you think?

Reply:
Hello @andyjpb,
Thanks for looking into it. A follow-up question just out of curiosity - what is the width of `intptr_t` and did you use `sizeof` to get the size of pointers?

If we don't have a type which has an architecture/platform specific width, I suggest that we stick to the original datatype (`uint32_t`). Even on platforms with low address width, we can use `uint32_t` (it is an overkill but safe).

I am just concerned since `intptr_t` might not be supported by all compilers. And this might break something in the long run. `uint32_t` in my opinion is safer.

Thoughts?

Reply:
@AniruddhaKanhere wrote:

>  And this might break something in the long run. uint32_t in my opinion is safer.

Isn't that what we had already? The function used to cast a pointer to a `uint32_t`, and store the 2 lower bits in `ulAlignBits`.
Now it will be cast to a variable which has the same size as a void pointer ( 16-bits on an ATmega ).

Reply:
Hi @AniruddhaKanhere,

The width of intptr_t is guaranteed to be an integer large enough to hold the address of a pointer on the target platform. I didn't use sizeof.
intptr_t is part of stdint.h, introduced in C99, so will be available everywhere the other platform-independent *_t types (uint32_t, etc) are available.

I'm happy to change it back if you wish but it will generate a warning on all platforms with pointers that are not 32bits. It could also cause problems on platforms where pointers are larger than 32bits.

So, yes, you're right: intptr_t might not be available on all compilers but it should certainly be available everywhere that has uint32_t.

Reply:
Thanks @htibosch and @andyjpb for your feedback/answers and patience with my suggestions!

While I agree that `intptr_t` is a better solution than `uint32_t`, there is no precedence of using this datatype anywhere in the code. I just don't want to set a precedence for various new datatypes in the source.

If @htibosch and @gkwicker are okay with this, I have no further objection :)

Reply:
> If @htibosch and @gkwicker are okay with this, I have no further objection :)

I am OK with the change. I do not have an overview of all available releases of `stdint.h`, but my feeling is that `intptr_t` is always part of it.
`uintptr_t` would be **better** though: pointers are also kind of unsigned, with `NULL` at the bottom.

( apart from this PR: there is more work to do on pointers: in every packet buffer space (`pucEthernetBuffer`), there is a hidden pointer, pointing to the owning `NetworkBufferDescriptor_t`.
This would also be a good place to adopt `uintptr_t`, in stead of just casting to `uint32_t` )